lib/engine/game/base.rb

```diff
@@ -764,6 +767,35 @@ def last_game_action_id
         @last_game_action_id || 0
       end
 
+      def previous_game_action_id_from(action_id)
+        if (action = @raw_all_actions.reverse.find { |a| a['id'] < action_id && !a['skip'] && a['type'] != 'message' })
```

Comment:
@raw.rverse.find {...}&.['id']

all of these can be cleaned up

Reply:
Ahh, cool. I didn't realize you can use "." with operator methods. Will clean up.

Reply:
Confirmed you can't use "&." with an operator method. I did find another hash method (fetch) that retrieved the value.